miv/io/data.py

```diff
@@ -182,55 +182,7 @@ def load_data(
             data = pickle.load(output_file, **pkl_kwargs)
         return data
 
-    @contextmanager
-    def load(self, start_at_zero: bool = False):
-        """
-        Context manager for loading data instantly.
-
-        Parameters
-        ----------
-        start_at_zero : bool
-            If set to True, time first timestamps will be shifted to zero. To achieve synchronized
-            timestamps with other recordings/events, set this to False.
-
-        Examples
-        --------
-            >>> data = Data(data_path)
-            >>> with data.load() as (signal, timestamps, sampling_rate):
-            ...     ...
-
-        Returns
-        -------
-        signal : SignalType, neo.core.AnalogSignal
-            The length of the first axis `signal.shape[0]` correspond to the length of the
-            signal, while second axis `signal.shape[1]` correspond to the number of channels.
-        timestamps : TimestampsType, numpy array
-        sampling_rate : float
-
-        Raises
-        ------
-        FileNotFoundError
-            If some key files are missing.
-
-        """
-        # TODO: Not sure this is safe implementation
-        if not self.check_path_validity():
-            raise FileNotFoundError("Data directory does not have all necessary files.")
-        timestamps, signal = None, None
-        try:
-            signal, timestamps, sampling_rate = next(
-                load_recording(
-                    self.data_path,
-                    self.masking_channel_set,
-                    start_at_zero=start_at_zero,
-                )
-            )
-            yield signal, timestamps, sampling_rate
-        finally:
-            del timestamps
-            del signal
-
-    def load_fragments(
+    def load(
```

Comment:
Removed the `contextmanager` for here too. Is this fine?

Reply:
Yes, its fine. `contextmanager` take care of object creation and deletion when we use `with` syntax. We are no longer using them.